x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/bulk_actions/context.ts

```diff
@@ -12,7 +12,7 @@ export const BulkActionsContext = createContext<
   [BulkActionsState, React.Dispatch<BulkActionsReducerAction>]
 >([
   {
-    rowSelection: new Set<number>(),
+    rowSelection: new Map<number, boolean>(),
```

Comment:
I'm wondering if we should just use an object here, works better with React reconciliation as we don't have to worry about accidentally not updating the reference and miss re-renders

Reply:
It should also make it more readable